app/audit.go

```diff
@@ -4,13 +4,79 @@
 package app
 
 import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/audit"
+	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
+const (
+	RestLevelID        = 240
+	RestContentLevelID = 241
+	RestPermsLevelID   = 242
+	CLILevelID         = 243
+)
+
+var (
+	RestLevel        = audit.Level{ID: RestLevelID, Name: "audit-rest", Stacktrace: false}
+	RestContentLevel = audit.Level{ID: RestContentLevelID, Name: "audit-rest-content", Stacktrace: false}
+	RestPermsLevel   = audit.Level{ID: RestPermsLevelID, Name: "audit-rest-perms", Stacktrace: false}
+	CLILevel         = audit.Level{ID: CLILevelID, Name: "audit-cli", Stacktrace: false}
+)
+
 func (a *App) GetAudits(userId string, limit int) (model.Audits, *model.AppError) {
 	return a.Srv().Store.Audit().Get(userId, 0, limit)
 }
 
 func (a *App) GetAuditsPage(userId string, page int, perPage int) (model.Audits, *model.AppError) {
 	return a.Srv().Store.Audit().Get(userId, page*perPage, perPage)
 }
+
+func (s *Server) configureAudit(adt *audit.Audit) {
+	adt.OnQueueFull = s.onAuditTargetQueueFull
+	adt.OnError = s.onAuditError
+
+	// For now we only support sending audit records to Syslog via TLS.
+	// See https://www.rsyslog.com/doc/v8-stable/tutorials/tls_cert_summary.html
+	if *s.Config().AuditSettings.Enabled {
+		IP := *s.Config().AuditSettings.IP
```

Comment:
Can we rename the config group to something like `ExperimentalAuditSettings` or something? Once we add something to that struct, we are stuck having to maintain it in code for ever, which is not going to be a nice experience, but will be less painful if it's prefixed `experimental`.

Reply:
To clarify "for every" - until a full deprecation cycle has completed, so 2 major versions. Feels like forever anyway.

Reply:
Makes sense.  Renamed to ExperimentalAuditSettings via https://github.com/mattermost/mattermost-server/pull/13967/commits/bfff7aca9dcd93b94af61ba2cff8e650651f7747